src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -164,6 +164,18 @@ private static async ValueTask<(bool, T?)> TryResolveBody<T>(HttpContext httpCon
         {
             return T.BindAsync(context, parameter);
         }
+
+        private static ValueTask<(bool, T?)> TryResolveJsonBodyOrServiceAsync<T>(HttpContext httpContext, bool isOptional, IServiceProviderIsService? serviceProviderIsService = null)
```

Comment:
This method should return a function that does body or JSON depending on the result of serviceProviderIsService

Reply:
Good point -- I'll do this in a follow-up...